modules/models/behavior/motion_primitives/continuous_actions.hpp

```diff
@@ -28,7 +30,7 @@ class BehaviorMPContinuousActions : public BehaviorMotionPrimitives {
     const ObservedWorld& observed_world);
 
   virtual MotionIdx GetNumMotionPrimitives(
-    const ObservedWorldPtr& observed_world) const {
```

Comment:
why not const?

Reply:
We want to avoid the recalculation of the valid motion primitives in `BehaviorMPMacroAction` so we cache the result locally:
https://github.com/bark-simulator/bark/blob/0eefd3587137099d0a21ddf8a3bf8b4b9b4b19df/modules/models/behavior/motion_primitives/macro_actions.cpp#L86